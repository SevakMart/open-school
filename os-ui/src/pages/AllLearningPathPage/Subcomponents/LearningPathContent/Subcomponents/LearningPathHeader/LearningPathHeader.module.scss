@import "../../../../../../styles/Mixins/FlexMixins.scss";
@import "../../../../../../styles/Mixins/FontMixin.scss";

.learningPathsHeader{
    @include flexStyle(row,center,space-between);
    margin-bottom:1.5rem;
    padding-right:6%;
    nav {
        @include flexStyle(row,center,space-between);
        width:30%;
        .activeNav{
            @include fontStyle('Montserrat',normal,600,1.25rem,1.5rem);
            color:#222427;
            border-bottom: 4px solid #848A9D;
            cursor:pointer;
        }
        .nonActiveNav{
            @include fontStyle('Montserrat',normal,600,1rem,1.5rem);
            color:#5E617B;
            cursor:pointer;
        }
    }
    .sortingContainer{
        @include fontStyle('Montserrat',normal,400,0.875rem,1.063rem);
        label {
            color:#848A9D;
        };
        select {
            background: none;
            outline: none;
            border:0px;
        }
    }
}
