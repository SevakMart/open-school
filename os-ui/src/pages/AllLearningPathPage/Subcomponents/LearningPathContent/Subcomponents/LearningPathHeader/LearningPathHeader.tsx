import React, {
  useState, useEffect, useRef, useContext,
} from 'react';
import { useDispatch } from 'react-redux';
import { useLocation, useNavigate } from 'react-router-dom';
import { addFilterParams } from '../../../../../../redux/Slices/AllLearningPathFilterParamsSlice';
import {
  ALL_LEARNING_PATHS, SAVED_LEARNING_PATHS, SORT_BY, DIFFICULTY, TITLE, RATING,
} from '../../../../../../constants/Strings';
import { headerTitleContext } from '../../../../../../contexts/Contexts';
import styles from './LearningPathHeader.module.scss';

const LearningPathHeader = ({ handleChangeHeader }:
  {handleChangeHeader:(headerTitle:string)=>void}) => {
  const location = useLocation();
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const focusedHeader = useContext(headerTitleContext);
  const params = new URLSearchParams(location.search);
  const selectionRef = useRef<HTMLSelectElement>(null);
  const [sortingFeature, setSortingFeature] = useState(RATING);

  const {
    learningPathsHeader, sortingContainer, activeNav, nonActiveNav,
  } = styles;

  const switchSortingFeatures = (e:React.SyntheticEvent) => {
    setSortingFeature((e.target as HTMLSelectElement).value);
  };

  useEffect(() => {
    if (params.has('sort'))setSortingFeature(params.get('sort') as string);
  }, []);

  useEffect(() => {
    if (focusedHeader === ALL_LEARNING_PATHS) {
      params.set('sort', sortingFeature);
      dispatch(addFilterParams({ sort: sortingFeature }));
      navigate(`/exploreLearningPaths?${params}`);
    }
  }, [sortingFeature]);

  return (
    <div className={learningPathsHeader} style={focusedHeader === ALL_LEARNING_PATHS ? { paddingLeft: '20%' } : { paddingLeft: '6%' }}>
      <nav>
        <p
          className={focusedHeader === ALL_LEARNING_PATHS ? activeNav : nonActiveNav}
          data-testid={ALL_LEARNING_PATHS}
          onClick={() => handleChangeHeader(SAVED_LEARNING_PATHS)}
        >
          {ALL_LEARNING_PATHS}

        </p>
        <p
          className={focusedHeader === SAVED_LEARNING_PATHS ? activeNav : nonActiveNav}
          data-testid={SAVED_LEARNING_PATHS}
          onClick={() => handleChangeHeader(ALL_LEARNING_PATHS)}
        >
          {SAVED_LEARNING_PATHS}

        </p>
      </nav>
      {focusedHeader === ALL_LEARNING_PATHS
      && (
      <div className={sortingContainer}>
        <label data-testid="sorting" htmlFor="sorting">{SORT_BY}</label>
        <select ref={selectionRef} value={sortingFeature} name="sorting" id="sorting" onChange={switchSortingFeatures}>
          <option value={RATING}>{RATING}</option>
          <option value={DIFFICULTY}>{DIFFICULTY}</option>
          <option value={TITLE}>{TITLE}</option>
        </select>
      </div>
      )}
    </div>
  );
};
export default LearningPathHeader;
