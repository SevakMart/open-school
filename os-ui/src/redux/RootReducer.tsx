import { combineReducers } from 'redux';
import chooseSubcategoryReducer from './Slices/ChoosSubcategorySlice';
import userInfoReducer from './Slices/loginUserSlice';
import allLearningPathFilterParamsReducer from './Slices/AllLearningPathFilterParamsSlice';
import forgotPasswordEmailReducer from './Slices/ForgotPasswordEmailSlice';
import signedUpUserIdReducer from './Slices/SignedUpUserIdSlice';
import suggestedCourseReducer from './Slices/SuggestedCourseSlice';
import userEnrolledCourseByCourseStatusReducer from './Slices/UserEnrolledCourseByCourseStatus';
import allLearningPathCoursesReducer from './Slices/AllLearningPathCourseSlice';
import allMentorsFilterParamsReducer from './Slices/AllMentorsFilterParamsSlice';
import allMentorsSliceReducer from './Slices/AllMentorsSlice';
import savedMentorsReducer from './Slices/SavedMentorsSlice';
import savedCourseReducer from './Slices/SavedLearningPathSlice';
import homepageMentorListReducer from './Slices/HomepageMentorSlice';
import portalStatusReducer from './Slices/PortalOpenStatus';

const rootReducer = combineReducers({
  portalStatus: portalStatusReducer,
  userInfo: userInfoReducer,
  chooseSubcategories: chooseSubcategoryReducer,
  filterParams: allLearningPathFilterParamsReducer,
  forgotPasswordEmail: forgotPasswordEmailReducer,
  signedUpUserId: signedUpUserIdReducer,
  suggestedCourse: suggestedCourseReducer,
  userEnrolledCourseByCourseStatus: userEnrolledCourseByCourseStatusReducer,
  allLearningPathCourses: allLearningPathCoursesReducer,
  allMentorsFilterParams: allMentorsFilterParamsReducer,
  allMentorsList: allMentorsSliceReducer,
  savedMentors: savedMentorsReducer,
  savedCourse: savedCourseReducer,
  homepageMentorList: homepageMentorListReducer,
});
export default rootReducer;
