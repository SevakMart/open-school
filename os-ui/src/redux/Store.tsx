import { configureStore } from '@reduxjs/toolkit';
import {
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import { createStateSyncMiddleware, initMessageListener } from 'redux-state-sync';
import rootReducer from './RootReducer';

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: {
      ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
    },
  }).concat(
    createStateSyncMiddleware({
      blacklist: ['persist/PERSIST', 'persist/REHYDRATE'],
    }),
  ),
});

initMessageListener(store);

export type RootState = ReturnType<typeof store.getState>
