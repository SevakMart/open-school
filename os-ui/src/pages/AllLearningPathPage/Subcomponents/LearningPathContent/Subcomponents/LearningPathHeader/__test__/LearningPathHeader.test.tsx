import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { store } from '../../../../../../../redux/Store';
import LearningPathHeader from '../LearningPathHeader';

const mockUseNavigate = jest.fn();
const mockUseLocation = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom') as any,
  useNavigate: () => mockUseNavigate,
  useLocation: () => mockUseLocation,
}));

describe('Create test cases for learningPathHeader Component', () => {
  test('Create a snapshot test', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <LearningPathHeader handleChangeHeader={() => null} />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  test('Test when we are on all learning path page,the app will render as expected', () => {
    expect.hasAssertions();
    render(
      <Provider store={store}>
        <LearningPathHeader handleChangeHeader={() => null} />
      </Provider>,
    );
    const sortingLabelElement = screen.queryByTestId('sorting');
    expect(sortingLabelElement).toBeInTheDocument();
  });
});
