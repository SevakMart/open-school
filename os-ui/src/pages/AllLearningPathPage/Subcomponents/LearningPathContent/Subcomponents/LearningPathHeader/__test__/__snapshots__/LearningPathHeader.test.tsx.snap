// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create test cases for learningPathHeader Component Create a snapshot test 1`] = `
<DocumentFragment>
  <div
    class="learningPathsHeader"
    style="padding-left: 20%;"
  >
    <nav>
      <p
        class="activeNav"
        data-testid="All Learning Paths"
      >
        All Learning Paths
      </p>
      <p
        class="nonActiveNav"
        data-testid="Saved Learning paths"
      >
        Saved Learning paths
      </p>
    </nav>
    <div
      class="sortingContainer"
    >
      <label
        data-testid="sorting"
        for="sorting"
      >
        Sort by
      </label>
      <select
        id="sorting"
        name="sorting"
      >
        <option
          value="Rating"
        >
          Rating
        </option>
        <option
          value="difficulty_id"
        >
          difficulty_id
        </option>
        <option
          value="Title"
        >
          Title
        </option>
      </select>
    </div>
  </div>
</DocumentFragment>
`;
